let CustomErrorHandler,JwtServices;_f4e‍.x([["default",()=>_f4e‍.o]]);_f4e‍.w("../services/CustomErrorHandler",[["default",["CustomErrorHandler"],function(v){CustomErrorHandler=v}]]);_f4e‍.w("../services/JwtServices",[["default",["JwtServices"],function(v){JwtServices=v}]]);


const auth = async (req, res, next) => {
    let authHeader = req.headers.authorization
    if(!authHeader) {
        return next(CustomErrorHandler.unAuthorized())
    }

    const token = authHeader.split(' ')[1]
    try {
        const {_id, role} = await JwtServices.verify(token)
        const user = {
            _id,
            role
        }
        req.user = user
        next()

    } catch (err) {
        return next(err)
    }

}

_f4e‍.d(auth);